File: android/src/main/kotlin/com/example/design_engine/layer2_core/dark/ESecondaryFixedDimDark.kt
package com.example.design_engine.layer2_core.dark
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object ESecondaryFixedDimDark {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyd30
            AppTheme.STANDARD -> R.color.cstd30
            AppTheme.BLUE -> R.color.cbd30
            AppTheme.GREEN -> R.color.cgd30
            AppTheme.RED -> R.color.crd30
            AppTheme.NOTHING -> R.color.cnod30
        }
    }
}

File: lib/layer2_core/dark/e_secondary_fixed_dim_dark.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/dark/system.dart';
import '../../layer1_raw/dark/standard.dart';
import '../../layer1_raw/dark/blue.dart';
import '../../layer1_raw/dark/green.dart';
import '../../layer1_raw/dark/red.dart';
import '../../layer1_raw/dark/nothing.dart';

class ESecondaryFixedDimDark {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyd30;
      case 'standard': return cstd30;
      case 'blue': return cbd30;
      case 'green': return cgd30;
      case 'red': return crd30;
      case 'nothing': return cnod30;
      default: return cstd30;
    }
  }
}
